File: codeforces/58A/py/58a.py
s = input()
dst = "hello"

id = 0
for i in range(len(s)):
    if s[i] == dst[id]:
        id += 1
    if id == len(dst):
        break

if id == len(dst):
    print("YES")
else:
    print("NO")

File: codeforces/58A/go/58a.go
package main

import "fmt"

func main() {
    var s string
    fmt.Scan(&s)

    var dst string = "hello"
    id := 0
    total := len(dst)
    for i := 0; i < len(s) && id < total; i++ {
        if s[i] == dst[id] {
            id++
        }
    }

    if id == total {
        fmt.Println("YES")
    } else {
        fmt.Println("NO")
    }
}
